### ISSUE TITLE
application.android.startActivity and foregroundActivity are undefined in unit tests

### ISSUE THREAD

**Issue by adrian-niculescu** [NativeScript/nativescript-cli#4922]
**Environment**
Provide version numbers for the following components (information can be retrieved by running `tns info` in your project folder or by inspecting the `package.json` of the project):
 - CLI: 6.0.2
 - Cross-platform modules: 6.0.4
 - **Android Runtime: 6.0.1**
 - iOS Runtime: 6.0.1

**Describe the bug**
`import * as application from 'tns-core-modules/application';`
`application.android.startActivity` and `application.android.foregroundActivity` are undefined in unit tests, not even after a few seconds after the app starts.

The problem is huge, because it affects all functionality that needs user permissions, as an activity is needed to request or check permissions on Android.

**To Reproduce**
Create an app from scratch:
`tns create MyApp`
`tns test init`

Write a test to check if application.android.startAcitivity or application.android.foregroundActivity exist.
Example: https://github.com/adrian-niculescu/NativeScriptUnitTestDemo/blob/master/app/tests/example.ts#L24

**Expected behavior**
`application.android.startActivity` and `application.android.foregroundActivity` to be defined.

**Sample project**
[Sample project](https://github.com/adrian-niculescu/NativeScriptUnitTestDemo)
It was created with:
`tns create MyApp`
`tns test init`
then added [this mocha test](https://github.com/adrian-niculescu/NativeScriptUnitTestDemo/blob/master/app/tests/example.ts#L24)

**Comment by DimitarTachev** [comment]
Hi @adrian-niculescu,

After cloning your repository and executing `tns test init` and `tns test android`, the provided test is passing and both of the activities are printed on the console.

![activitiesTest](https://user-images.githubusercontent.com/1865068/62703797-a56b2f00-b9f2-11e9-9448-15d97f1e8cdd.png)

Maybe you have some outdated git ignored files locally. Could you try cloning the provided repository in a new folder and running the test again?

**Comment by adrian-niculescu** [comment]
Following the exact steps I replicated this on 3 different machines: 2 with MacOS  and 1 Linux (CentOS 7).
Additional info about the environement
```
[centos7]$ node --version
v10.15.1

[macos]$ node --version
node --version
v10.16.0

$ npm --version
6.10.3
```
![Screenshot 2019-08-08 at 18 11 06](https://user-images.githubusercontent.com/15037449/62714881-0d784000-ba08-11e9-908d-0c343fda6044.png)

**Comment by DimitarTachev** [comment]
@adrian-niculescu

Thanks for the additional details.

We were able to reproduce the issue. It's caused by a wrong version of `karma-webpack` (it should be downgraded from `4.0.2` to `3.0.5`). You could take a look at [this issue](https://github.com/NativeScript/nativescript-cli/issues/4746) for more details.

The issue is already fixed in `nativescript@6.0.3`, in order to fix it on your side, you could follow the steps below:
1) `npm i nativescript@latest -g`
2) `tns test init` (this will set the proper `karma-webpack` version)

**Comment by adrian-niculescu** [comment]
Changing the version of `karma-webpack` to `3.0.5` solves the issue.
Thank you for your response.
